migrate.py

```diff
@@ -1198,6 +1146,57 @@ def assemble_collections(checkout_path, spec, args, target_github_org):
             REMOVE = set()
 
 
+def galaxy_metadata_init(collection, namespace, target_github_org):
+    return {
+        'namespace': namespace,
+        'name': collection,
+        'version': '1.0.0',  # TODO: add to spec, args?
+        'readme': None,
+        'authors': None,
+        'description': None,
+        'license': None,
+        'license_file': None,
+        'tags': None,
+        'dependencies': {},
+        'repository': f'git@github.com:{target_github_org}/{namespace}.{collection}.git',
+        'documentation': None,
+        'homepage': None,
+        'issues': None
+    }
+
+
+def process_symlink(namespace, collection, plugin_type, dest, src, spec):
+    real_src = os.readlink(src)
+
+    # remove destination if it already exists
+    if os.path.exists(dest):
+        # NOTE: not atomic but should not matter in our script
+        logger.warning('Removed "%s" as it is target for symlink of "%s"' % (dest, src))
+        os.remove(dest)
+
+    if real_src.startswith('../'):
+        target = real_src[3:]
+        found = False
+        for k in spec[namespace][collection][plugin_type]:
+            if k.endswith(target):
+                found = True
+                break
+
+        if found:
+            if plugin_type == 'module_utils':
+                target = real_src
+            else:
+                target = os.path.basename(real_src)
+            ret = os.getcwd()
+            os.chdir(os.path.dirname(dest))
+            os.symlink(os.path.basename(target), os.path.basename(dest))
+            os.chdir(ret)
```

Comment:
This requires try/finally. Otherwise, IO errors may cause the script to end up in unexpected states.
The best way to handle this semantically is to use a helper CM.

```suggestion
            with working_directory(os.path.dirname(dest)):
                os.symlink(os.path.basename(target), os.path.basename(dest))
```

Such helper would look like:
```python
@contextlib.contextmanager
def working_directory(target_dir):
    current_working_dir = os.getcwd()
    os.chdir(target_dir)
    try:
        yield os.getcwd()
    finally:
        os.chdir(current_working_dir)
```